sec_certs_page/dashboard/manager.py

```diff
@@ -0,0 +1,219 @@
+            CCValidityDuration(
+                graph_id="cc-validity-duration",
+                data_service=self.data_service,
+                config=validity_duration_config,
+            ),
+        ]
+
+        for chart in charts:
+            cc_chart_registry.register(chart)
+
+    def _register_fips_charts(self) -> None:
+        pass
+
+    def get_dashboard_names(
+        self,
+        user_id: str,
+        dataset_type: CollectionName,
+    ) -> list[dict[str, str]]:
+        """
+        Get dashboard names for dropdown population (INIT-1 lazy loading).
+
+        Returns minimal data: dashboard_id, name, is_default flag.
+        """
+        return self.repository.get_names_by_user(user_id, dataset_type)
+
+    def load_dashboard_with_charts(
+        self,
+        dashboard_id: str,
+    ) -> tuple[Dashboard | None, list]:
+        """
+        Load dashboard and instantiate BaseChart components from stored configs.
+
+        Implements INIT-2 callback reconstruction by using ChartFactory
+        to create runtime chart instances from serialized Chart configs.
+
+        :return: Tuple of (Dashboard, list of BaseChart instances)
+        """
+        dashboard = self.repository.get_by_id(dashboard_id)
+        if dashboard is None:
+            return None, []
+
+        chart_instances = []
+        for chart_config in dashboard.charts:
+            try:
+                chart_instance = ChartFactory.create_chart(chart_config, self.data_service)
+                chart_instances.append(chart_instance)
+            except ValueError as e:
+                print(f"Warning: Could not instantiate chart {chart_config.name}: {e}")
+                continue
+
+        return dashboard, chart_instances
+
+    def load_default_dashboard(
+        self,
+        user_id: str,
+        dataset_type: CollectionName,
+    ) -> tuple[Dashboard | None, list]:
+        """
+        Load user's default dashboard for INIT-3 first access.
+
+        :return: Tuple of (Dashboard, list of BaseChart instances) or (None, [])
+        """
+        dashboard = self.repository.get_default(user_id, dataset_type)
+        if dashboard is None:
+            return None, []
+
+        chart_instances = []
+        for chart_config in dashboard.charts:
+            try:
+                chart_instance = ChartFactory.create_chart(chart_config, self.data_service)
+                chart_instances.append(chart_instance)
+            except ValueError:
+                continue
+
+        return dashboard, chart_instances
+
+    def get_predefined_chart_configs(self, dataset_type: CollectionName) -> list[Chart]:
+        """
+        Get predefined chart configurations for 'Load Predefined' option.
+
+        Returns Chart dataclass instances (not BaseChart) for serialization.
+        """
+        registry = self.chart_registries.get(dataset_type)
+        if not registry:
+            return []
+
+        return [chart.config for chart in registry]
```

Comment:
This will also get the custom chart configs if they were previously added to the registry.  Hence my comments about not using that global state.